### PR TITLE
Add validate command to zed

### PR DESCRIPTION
**Pull request by josephschorr**
The validate command allows a Playground-style YAML file containing schema, relationships, assertions and expected relations to be validated locally from the CLI

Fixes https://github.com/authzed/spicedb/issues/290

### CODE REVIEW

File: cmd/zed/validate.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+var (
+	success                = lipgloss.NewStyle().Bold(true).Foreground(lipgloss.Color("#00ff00")).Render("success!")
+	errorPrefix            = lipgloss.NewStyle().Bold(true).Foreground(lipgloss.Color("#ff0000")).Render("error: ")
+	errorMessageStyle      = lipgloss.NewStyle().Bold(true).Width(80)
+	linePrefixStyle        = lipgloss.NewStyle().Foreground(lipgloss.Color("#0000ff"))
+	highlightedSourceStyle = lipgloss.NewStyle().Foreground(lipgloss.Color("#ff0000"))
+	highlightedLineStyle   = lipgloss.NewStyle().Foreground(lipgloss.Color("#ff0000"))
+	codeStyle              = lipgloss.NewStyle().Foreground(lipgloss.Color("#555555"))
+	highlightedCodeStyle   = lipgloss.NewStyle().Foreground(lipgloss.Color("#ffffff"))
```
Comment (jzelinskie) [inline cmd/zed/validate.go:31]: If you're only using basic colors, use the ANSI colors: https://github.com/charmbracelet/lipgloss#ansi-256-colors-8-bit

By hardcoding it to be hex, it just makes it really hard to read for people with different color-schemes.

File: cmd/zed/validate.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+	// Run expected relations.
+	_, erDevErrs, rerr := development.RunValidation(ctx, devCtx, parsed.ExpectedRelations)
+	if rerr != nil {
+		return rerr
+	}
+	if erDevErrs != nil {
+		outputDeveloperErrors(validateContents, erDevErrs, p.Validation.Line-2) // -1 for the key name and -1 since 1-indexed
+	}
+
+	fmt.Println(success)
```
[the remark this answers is not in the document]
Comment (josephschorr) [reply cmd/zed/validate.go:123]: I think it is desirable, since its mostly designed for people to work with.

File: cmd/zed/validate.go (hunk 1 above)
Comment (josephschorr) [reply cmd/zed/validate.go:32]: ANSI color codes seem even more opaque though

Comment (jzelinskie) [reply cmd/zed/validate.go:32]: ANSI color codes use the color theme your terminal is configured to use. Hex forces absolute colors that can be entirely illegible depending on what you've configured.

Comment (josephschorr) [reply cmd/zed/validate.go]: Done

File: cmd/zed/validate.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+	"github.com/charmbracelet/lipgloss"
+
+	"github.com/authzed/zed/internal/decode"
+)
+
+var (
+	success                = lipgloss.NewStyle().Bold(true).Foreground(lipgloss.Color("10")).Render("success!")
+	errorPrefix            = lipgloss.NewStyle().Bold(true).Foreground(lipgloss.Color("9")).Render("error: ")
+	errorMessageStyle      = lipgloss.NewStyle().Bold(true).Width(80)
+	linePrefixStyle        = lipgloss.NewStyle().Foreground(lipgloss.Color("19"))
```
Comment (jzelinskie) [inline cmd/zed/validate.go:28]: Can this just be the same as the regular code style ("8")? It's a bit distracting as blue. If you want to keep it blue, you can use "12".

File: cmd/zed/validate.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jzelinskie) [reply cmd/zed/validate.go:123]: Maybe we can output the number of validated assertions, relations, etc...

File: cmd/zed/validate.go (hunk 3 above)
Comment (josephschorr) [reply cmd/zed/validate.go:28]: Used 12

**Comment by josephschorr** [comment]
Updated

**Review by jzelinskie — APPROVED** [review]

**Pull request merged**